opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -275,6 +289,30 @@ def set_status(self, status: trace_api.Status) -> None:
             return
         self.status = status
 
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
+        """Ends context manager and calls `end` on the `Span`."""
+
+        if self.status is None and self._auto_update_status:
+            if exc_val is not None:
+
+                if isinstance(exc_val, StatusError):
+                    self.set_status(
+                        Status(exc_val.canonical_code, exc_val.description)
+                    )
+
+                else:
+                    self.set_status(Status(StatusCanonicalCode.UNKNOWN))
+
+            else:
+                self.set_status(Status(StatusCanonicalCode.OK))
```

Comment:
I think that may be against [this](https://github.com/open-telemetry/opentelemetry-python/issues/292#issuecomment-554484563) request. I also think is cleaner if we put the automatic status setting mechanisms *only* in the context manager. In this way, it is possible to simply state that if the context manager is not used, then the user has full control and is the responsibility of the user to handle all the starting, ending and status setting of the span manually.

Reply:
By the way, should `__enter__` call `self.start`?

Reply:
No, by the time `__enter__` is called, the span will always be started already (we don't have unstarted spans anymore).

Reply:
> I think that may be against this request.

I think there is a slight misunderstanding here. I only meant the `else:` part should be moved to `end()`. Because right now there is actually a bug:

```python
my_span = tracer.start_span('mycoolspanname')
my_span.end() # This will cause a span with `status == None` to be sent to the exporter, which is invalid
```

Reply:
Ok, moved this to end inside an `if` that guards against overriding a status previously set by the user.